cmd/reconciler/ip.go

```diff
@@ -1,40 +1,46 @@
-package main
+// package main
+package reconciler
 
 import (
 	"context"
+	"errors"
 	"flag"
-	"os"
+	"time"
 
 	"github.com/k8snetworkplumbingwg/whereabouts/pkg/logging"
 	"github.com/k8snetworkplumbingwg/whereabouts/pkg/reconciler"
 )
 
 const defaultReconcilerTimeout = 30
 
-func main() {
-	kubeConfigFile := flag.String("kubeconfig", "", "the path to the Kubernetes configuration file")
-	logLevel := flag.String("log-level", "error", "the logging level for the `ip-reconciler` app. Valid values are: \"debug\", \"verbose\", \"error\", and \"panic\".")
-	reconcilerTimeout := flag.Int("timeout", defaultReconcilerTimeout, "the value for a request timeout in seconds.")
-	flag.Parse()
+func InvokeIPReconciler(returnErr chan error) {
+	go func() {
+		doInvokeIPReconciler(returnErr)
+	}()
+	select {
+	case <-time.After(5 * time.Second):
+		returnErr <- errors.New("timed out (this is the 5 sec test timeout, nbd)")
+		return
```

Comment:
you don't need to do this. You should pass a `context.WithTimeout` below, in line https://github.com/k8snetworkplumbingwg/whereabouts/pull/218/files#diff-99080d2a71bc34fba51f6b2bb1766239380ae51bdf70aa1496fa73b5ffdbbe79R31

I also suspect that the issue you're facing comes from spawning 2 go routines
 - first in https://github.com/k8snetworkplumbingwg/whereabouts/pull/218/files#diff-dbddff16de1ff73a326e91a361beb31cae41142fa4b9e85d5e759575af2a56bdR98
 - second in https://github.com/k8snetworkplumbingwg/whereabouts/pull/218/files#diff-99080d2a71bc34fba51f6b2bb1766239380ae51bdf70aa1496fa73b5ffdbbe79R17

but I am not 100% sure of this second claim.